src/task/CamelKTaskCompletionItemProvider.ts

```diff
@@ -22,65 +22,66 @@ import { TraitManager } from './TraitManager';
 
 export class CamelKTaskCompletionItemProvider implements vscode.CompletionItemProvider {
 
-    provideCompletionItems(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken, context: vscode.CompletionContext): vscode.ProviderResult<vscode.CompletionItem[] | vscode.CompletionList> {
-        return this.provideCompletionItemsForText(document.getText(), document.offsetAt(position));
-    }
+	provideCompletionItems(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken, context: vscode.CompletionContext): vscode.ProviderResult<vscode.CompletionItem[] | vscode.CompletionList> {
+		return this.provideCompletionItemsForText(document.getText(), document.offsetAt(position));
+	}
 
-    public async provideCompletionItemsForText(text: string, offset: number): Promise<vscode.CompletionItem[]> {
-        let node = jsonparser.findNodeAtOffset(jsonparser.parseTree(text), offset, false);
-        let completions: vscode.CompletionItem[] = [];
-        if (node !== undefined) {
-            if (this.isInTasksArray(node)) {
-                let completionBasic: vscode.CompletionItem = {
-                    label: 'Camel K basic development mode',
-                    insertText:
-`{
+	public async provideCompletionItemsForText(text: string, offset: number): Promise<vscode.CompletionItem[]> {
+		const globalNode = jsonparser.parseTree(text);
+		let node = jsonparser.findNodeAtOffset(globalNode, offset, false);
+		let completions: vscode.CompletionItem[] = [];
+		if (node !== undefined) {
+			if (this.isInTasksArray(node)) {
+				let completionBasic: vscode.CompletionItem = {
+					label: 'Camel K basic development mode',
+					insertText:
+						`{
     "label": "Start in dev mode Camel K integration opened in active editor",
     "type": "camel-k",
     "dev": true,
     "file": "\${file}",
     "problemMatcher": []
 }`
-                };
-                completions.push(completionBasic);
-            } else if(this.isInTraitsArray(node)) {
-                let traitCompletions: vscode.CompletionItem[] = await TraitManager.provideAvailableTraits();
-                completions = completions.concat(traitCompletions);
-            }
-        }
-        return Promise.resolve(completions);
-    }
+				};
+				completions.push(completionBasic);
+			} else if (this.isInTraitsArray(node)) {
+				let traitCompletions: vscode.CompletionItem[] = await TraitManager.provideAvailableTraits();
```

Comment:
traitCompletions will not be changed in that function so it could be made a const instead.